src/base/bittorrent/cachestatus.cpp

```diff
@@ -50,7 +50,11 @@ qreal CacheStatus::readRatio() const
 
 int CacheStatus::jobQueueLength() const
 {
+#if LIBTORRENT_VERSION_NUM < 10100
     return m_nativeStatus.job_queue_length;
+#else
+    return m_nativeStatus.queued_jobs;
```

Comment:
One major feature I've been working on in the last few years is to provide more consistent and efficient reporting mechanism for data out of libtorrent. The motivating reason was originally to provide performance counters to instrument and measure performance of large scale systems to tweak settings. In 1.0, this counters are available as a separate build of libtorrent which makes it not very convenient, because you need to rebuild when you suspect a performance issue.

The new system is flexible in the sense that new counters can be added without breaking the ABI and counters are efficiently provided in their raw form to clients and if there is to be a log of them to be analyzed, the client may log them to a file. They can also be disabled. I would imagine that for qBT to retrieve counters via this new API, there would need to be a little bit of refactoring. They are requested via a function on session and delivered as an alert.

( see https://github.com/arvidn/libtorrent/blob/master/docs/manual.rst#session-statistics and https://github.com/arvidn/libtorrent/blob/master/src/session_stats.cpp#L48 )

I tried to kill the cache_status entirely, but it is the only way to get a snapshot of the actual pieces in the cache, so I left that un-deprecated.

The session_settings value is not a performance counter, it's the setting and it will only change if the client changes it.